infra-in-containers/resources/deploy.yml

```diff
@@ -0,0 +1,135 @@
+      default: "jiunnyuan@commandocloudlet.com"
+      private: no
+    - name: "email_password"
+      prompt: "account password"
+      private: yes
+      confirm: yes
+    - name: "relay_host"
+      prompt: "What is your email relay host?"
+      default: "smtp.sendgrid.net"
+      private: no
+    - name: "relay_port"
+      prompt: "What is your email relay port?"
+      default: "587"
+      private: no
+    - name: "relay_username"
+      prompt: "What is your email relay username?"
+      default: "apikey"
+      private: no
+    - name: "relay_key"
+      prompt: "What is your email relay password?"
+      default: "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"
+      private: no
+
+  tasks:
+    - name: Ensure {{base_path}} exists
+      file: path={{base_path}} state=directory
+    - name: "download docker-compose.yml"
+      get_url:
+        url: https://raw.githubusercontent.com/CloudCommandos/missions/CC/infra-in-containers/resources/docker-compose.yml
+        dest: "{{base_path}}/docker-compose.yml"
+        mode: 0777
+        force: yes
+    - name: "download .env"
+      get_url:
+        url: https://raw.githubusercontent.com/CloudCommandos/missions/CC/infra-in-containers/resources/env.txt
+        dest: "{{base_path}}/.env"
+        mode: 0777
+        force: yes
+    - name: "download setup.sh"
+      get_url:
+        url: https://raw.githubusercontent.com/CloudCommandos/missions/CC/infra-in-containers/resources/setup.sh
+        dest: "{{base_path}}/setup.sh"
+        mode: 0777
+        force: yes
+
+    - name: "edit domain name in .env"
+      lineinfile:
+        dest: "{{base_path}}/.env"
+        regexp: '^DOMAINNAME='
+        line: 'DOMAINNAME={{domain}}'
+    - name: install dependencies
+      apt:
+        name:
+            - apt-transport-https
+            - ca-certificates
+            - curl
+            - gnupg2
+            - software-properties-common
+            - python-pip
+            - python-apt
+            - openssl
+        state: present
+    - name: install pexpect
+      pip:
+        name: pexpect
+    - name: adding apt-key for docker
+      apt_key:
+        url: https://download.docker.com/linux/debian/gpg
+        state: present
+    - name: adding docker repo list
+      apt_repository:
+        repo: deb [arch=amd64] https://download.docker.com/linux/debian stretch stable
+        state: present
+    - name: install docker-ce and docker-compose
+      apt:
+        name:
+            - docker-compose
+            - docker-ce
+        state: present
+    - name: create directory
+      file:
+        path: "{{base_path}}/ssl"
+        state: directory
+    - name: generate a self signed OpenSSL certificate
+      expect:
+        command: openssl req -new -newkey rsa:4096 -x509 -sha256 -days 365 -nodes -out {{base_path}}/ssl/ssl.crt -keyout {{base_path}}/ssl/ssl.key
```

Comment:
Ansible has an openssl_certificate module. Alternatively, cfssl or EasyRSA might be a better way to manage your certs.

Reply:
I did try to use the openssl_certificate module. However, there seems to be an error when I was trying to run the module. Thus, the command module was used.

I managed to use ansible script to use EasyRSA to generate the certificate. The script to generate certificate using EasyRSA is shown below. Is this the correct way to use EasyRSA? It seems like its more complicated than using openssl.

```
tasks:
        - name: install dependencies
          apt:
                name:
                    - openvpn
                    - python-pip
                state: present
        - name: install pexpect
          pip:
                name: pexpect
        - name: download EasyRSA
          get_url:
                   url: https://github.com/OpenVPN/easy-rsa/releases/download/v$
                   dest: /root
        - name: decompress EasyRSA
          unarchive:
            src: /root/EasyRSA-3.0.4.tgz
            dest: /root
            remote_src: yes
        - name: copy vars from vars.example
          copy:
            src: /root/EasyRSA-3.0.4/vars.example
            dest: /root/EasyRSA-3.0.4/vars
        - name: uncomment on vars file
          lineinfile:
            path: /root/EasyRSA-3.0.4/vars
            regexp: '#set_var EASYRSA_DN'
            line: 'set_var EASYRSA_DN   "cn_only"'
        - name: initialise EasyRSA
          expect:
            command: ./easyrsa init-pki
            responses:
                 "Confirm removal:" :  "yes"
          args:
            chdir: "/root/EasyRSA-3.0.4/"
        - name: generate self signed certificate
          expect:
            command: ./easyrsa build-ca nopass
            responses:
                 "Generating" : ""
          args:
            chdir: "/root/EasyRSA-3.0.4/"
        - name: copy the certificate and key to the ssl folder
          copy:
            src: "{{ item.src }}"
            dest: "{{ item.dest}}"
          with_items:
            - { src: '/root/EasyRSA-3.0.4/pki/ca.crt', dest: '/root/dockerproj/docker-mailserver/ssl/ssl.crt' }
            - { src: '/root/EasyRSA-3.0.4/pki/private/ca.key', dest: '/root/dockerproj/docker-mailserver/ssl/ssl.key' }
```